src/rgw/rgw_cache.cc

```diff
@@ -27,8 +27,7 @@ int ObjectCache::get(const string& name, ObjectCacheInfo& info, uint32_t mask, r
   if (expiry.count() &&
        (ceph::coarse_mono_clock::now() - iter->second.info.time_added) > expiry) {
     ldout(cct, 10) << "cache get: name=" << name << " : expiry miss" << dendl;
-    lock.unlock();
-    lock.lock();
+    std::unique_lock wl{lock};
```

Comment:
the `std::shared_lock l{lock};` still thinks it holds this lock, so will call `lock.unlock_shared()` on its destruction. shouldn't this call `l.unlock()` before reacquiring the unique lock?

Reply:
You're right. I noticed some similar stuff in other parts of Ceph, where we take a lock guard and then call unlock and lock on the mutex.

An artifact from Mutex::Locker not having lock/unlock methods, but it would be worthwhile to find all those instances and Destroy Them.

Reply:
do we need the shared_lock for the function scope at all, or is it only necessary for the map.find() in which case we can scope the shared lock to only the above cache_map.find?

Reply:
err scratch that, pushed the changes, if they look okay I'll squash them